test/synthetic-asset.js

```diff
@@ -0,0 +1,183 @@
+      await expect(syntheticTSLAContract.connect(hub).burn(owner.address, 100))
+        .to.be.revertedWith("The user has not enough assets");
+    });
+  });
+
+  /**************************************** Getters ****************************************/
+
+  describe("GetAmountEligibleToBurn", () => {
+    it("Should return the amount of synthetic asset the user is elligible to burn", async () => {
+      // Check that the owner's synthetic TSLA balance is empty.
+      let ownerBalance = await syntheticTSLAContract.balanceOf(owner.address);
+      expect(ownerBalance).to.equal(0);
+
+      // Check that the owner is elligible to burn zero synthetic TSLA assets.
+      let amountElligibleToBurn = await syntheticTSLAContract.getAmountEligibleToBurn(owner.address);
+      expect(amountElligibleToBurn).to.equal(0);
+
+      // Mint 100 synthetic TSLA assets to the owner's wallet.
+      let txn = await syntheticTSLAContract.connect(hub).mint(owner.address, 100);
+      await txn.wait();
+
+      // Check that the owner is elligible to burn 100 synthetic TSLA assets.
+      amountElligibleToBurn = await syntheticTSLAContract.getAmountEligibleToBurn(owner.address);
+      expect(amountElligibleToBurn).to.equal(100);
+
+      // Check that the owner's synthetic TSLA balance is equal to 100.
+      ownerBalance = await syntheticTSLAContract.balanceOf(owner.address);
+      expect(ownerBalance).to.equal(100);
+
+      // Burn 40 synthetic TSLA assets from the owner's wallet.
+      await syntheticTSLAContract.connect(hub).burn(owner.address, 40);
+      await txn.wait();
+
+      // Check that the owner is elligible to burn 60 synthetic TSLA assets.
+      amountElligibleToBurn = await syntheticTSLAContract.getAmountEligibleToBurn(owner.address);
+      expect(amountElligibleToBurn).to.equal(60);
+
+      // It should fail when trying to get the elligible synthetic asset to burn of a null address.
+      await expect(syntheticTSLAContract.connect(hub).getAmountEligibleToBurn(ethers.constants.AddressZero))
+        .to.be.revertedWith("The address parameter cannot be null");
+    });
+  });
+
+  /**************************************** Setters ****************************************/
+
+  describe("SetAmountEligibleToBurn", () => {
+    it("Should update the amount of synthetic asset the user is elligible to burn", async () => {
+      // Check that the owner's synthetic TSLA balance is empty.
+      let ownerBalance = await syntheticTSLAContract.balanceOf(owner.address);
+      expect(ownerBalance).to.equal(0);
+
+      // Check that the owner is elligible to burn zero synthetic TSLA assets.
+      let amountElligibleToBurn = await syntheticTSLAContract.getAmountEligibleToBurn(owner.address);
+      expect(amountElligibleToBurn).to.equal(0);
+
+      // Update the owner's synthetic TSLA balance to 100.
+      const txn = await syntheticTSLAContract.connect(hub).setAmountEligibleToBurn(owner.address, 100);
+      await txn.wait();
```

Comment:
this is btw not the real way to use this function, set AmountEligibleToBurn is only used for liquidation, I know this is just a test, but to make things clear

Reply:
Yep you're right, i'll update my comment!